test/migrator_test.dart

```diff
@@ -0,0 +1,114 @@
+// Copyright 2018 Google LLC
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'dart:io';
+
+import 'package:path/path.dart' as p;
+import 'package:sass_module_migrator/src/migrator.dart';
+import 'package:sass_module_migrator/src/stylesheet_api.dart';
+import 'package:test/test.dart';
+
+void main() {
+  testHrx("variables");
+  testHrx("subdirectories");
+  testHrx("functions");
+  testHrx("mixins");
+}
+
+const testDirectory = "/test/";
+
+class TestMigrator extends Migrator {
+  Map<String, String> testFiles = {};
+
+  TestMigrator(this.testFiles);
+
+  @override
+  String loadFile(Path path) =>
+      testFiles[p.relative(path.path, from: testDirectory)];
+
+  @override
+  bool exists(Path path) =>
+      testFiles.containsKey(p.relative(path.path, from: testDirectory));
+
+  @override
+  String entrypointDirectory = testDirectory;
+
+  final List<String> logged = [];
+
+  @override
+  void log(String text) => logged.add(text);
+}
+
+class HrxTestFiles {
+  Map<String, String> testFiles = {};
+  Map<String, String> expectedOutput = {};
+  Map<String, List<String>> recursiveManifest = {};
+
+  HrxTestFiles(String hrxName) {
+    var hrxText = File("test/migrations/$hrxName.hrx").readAsStringSync();
+    // TODO(jathak): Replace this with an actual HRX parser.
+    String filename;
+    String contents;
+    for (String line in hrxText.substring(0, hrxText.length - 1).split("\n")) {
```

Comment:
```suggestion
    for (var line in hrxText.substring(0, hrxText.length - 1).split("\n")) {
```

[Avoid type annotating initialized local variables](https://www.dartlang.org/guides/language/effective-dart/design#avoid-type-annotating-initialized-local-variables).